### PR TITLE
allow --rpc-file option to change default value 

### PR DESCRIPTION
**Pull request by Saibato**
Stumbled on this when apache PHP  was not able to fetch return values from lightning-cli call 
Before that it always crashed when i used other rpc-file names then lightning-rpc.
Please review.

### CODE REVIEW

File: common/configdir.c [hunk 1]
```diff
@@ -22,11 +22,21 @@ static char *default_configdir(const tal_t *ctx)
 	return path;
 }
 
+static char *default_rpc_file(const tal_t *ctx)
+{
+	char *path;
+	
+    path = path_basename(ctx, "lightning-rpc");
+
+	return path;
```
Comment (cdecker) [inline common/configdir.c]: Inconsistent indentation, please use tabs (https://www.youtube.com/watch?v=SsoOG6ZeyUI) :wink:

**Comment by cdecker** [comment]
If I understand this correctly you're removing the `$HOME` resolution with the current directory, since we'll have changed into that directory by the time this code runs? It doesn't seem to change much at all, except the removal of the resolution. Can you share what the problem was?

**Comment by Saibato** [comment]
No even $home etc was not there before.
My first fix did set it to home env etc but then I realized that will not put it in the desired lightning-dir when set so I the last commit with only setting dir by path_basename can work.

cli just did not even start when any option --rpc-file was given.

File: common/configdir.c (hunk 1 above)
Comment (Saibato) [reply common/configdir.c]: ^^ removed some whitespace valgrind lammers..

The issue with this rpc-file came up when i want to call the lightning-cli  with options from PHP 
this still does not work.
Do i have to open the unix socket directly in a PHP script?

**Comment by Saibato** [comment]
cli/lightning-cli --rpc-file anyname
Abgebrochen

so now with this patch it can be set without crash to anyname :-)
 
a  proofofwork  tor onion site r4hwyyyrr7pd6nxh.onion:80
connects now with lightning-cli over  PHP  
no idea if this works stable its on  mainnet ^^

File: common/configdir.c [hunk 2]
```diff
@@ -22,15 +22,22 @@ static char *default_configdir(const tal_t *ctx)
 	return path;
 }
 
+static char *default_rpc_file(const tal_t *ctx)
+{
+	char *path;
+	path = path_basename(ctx, "lightning-rpc");
+	return path;
+}
```
Comment (rustyrussell) [inline common/configdir.c]: Ah!  I see, thanks for bug report!

--rpc-file= was trying to free the old string, but it was a literal.  The simplest fix here is to change below:

File: common/configdir.c [hunk 3]
```diff
@@ -22,15 +22,22 @@ static char *default_configdir(const tal_t *ctx)
...
+
 void configdir_register_opts(const tal_t *ctx,
-			     char **configdir, char **rpc_filename)
+				 char **configdir, char **rpc_filename)
 {
 	*configdir = default_configdir(ctx);
-	*rpc_filename = "lightning-rpc";
+	*rpc_filename = default_rpc_file(ctx);
```
Comment (rustyrussell) [inline common/configdir.c]: `*rpc_filename = tal_strdup(ctx, "lightning-rpc");`

That's the one-line fix :)

**Comment by Saibato** [comment]
sorry for the name spelling but my IQ 's let me read things that are not there yet ;-)

**Comment by rustyrussell** [comment]
Hi!  Somehow this has become a 10 commit monster, for what is really a 1 line patch :)

I can't even get git to squash it down to a single commit :(

Can you please close this, and create a single commit against master which just changes to tal_strdup()?  Or just do that then force-push over this branch.  Then I can apply it...

Thanks, and sorry for the confusion!

**Pull request closed without merging**